tools/src/bin/crossref.rs

```diff
@@ -225,21 +225,28 @@ fn main() {
 
         let _ = outputf.write_all(format!("{}\n{}\n", id, kindmap.to_string()).as_bytes());
 
-        if id_data.contains_key(&AnalysisKind::Def) {
-            let defs = id_data.get(&AnalysisKind::Def).unwrap();
-            if defs.len() == 1 {
-                for (path, results) in defs {
-                    if results.len() == 1 {
-                        let mut v = Vec::new();
-                        v.push(id.to_json());
-                        v.push(path.to_json());
-                        v.push(results[0].lineno.to_json());
-                        let pretty = pretty_table.get(&id).unwrap();
-                        v.push(pretty.to_json());
-                        jumps.push(Json::Array(v));
+        {
+            let mut add_jumps = |kind: &AnalysisKind, kindstr: &str| {
+                if id_data.contains_key(kind) {
+                    let defs = id_data.get(kind).unwrap();
```

Comment:
Instead of doing the contains_key check followed by get(..).unwrap(), you could combine them using the map function on Option:

```rust
let mut add_jumps = |kind: &AnalysisKind, kindstr: &str| {
    id_data.get(kind).map(|defs| {
        if defs.len() ...
    });
};